Assembler coding standards.md

```diff
@@ -0,0 +1,68 @@
+# Assembler coding standards
+
+## General principles
+1. Use simple, easy-to-understand logic
+2. Use mostly simple instructions
+3. Do not use the location counter in branch instructions (e.g. J *+6)  
+4. Use subroutines to break up your code into manageable pieces 
+5. Use structured programming concepts and structured programming macros  
+   1. Align op-codes and operands in a meaningful way 
+6. Use DSECTs to map data areas 
+7. Use efficient algorithms and describe in a comment block what you're doing   
+8. Request and check feedback from macro instructions 
+9. Design internal tables to expand easily 
+10. Provide meaningful error messages
+11. Use only standard interfaces 
+    1. Use macros to get to internals
+    2. Don't use constructs that are not defined as a programming interface
+    3. Tell the macros your environment (and don't lie) 
+       1. The macros change depending on SYSSTATE  
+12. Use standard entry/exits whenever possible
+    1.  Use the standard linkage conventions defined in the High-Level Assembler User Guide 
+    2. Clean up after yourself  
+13. Let the Assembler determine lengths (equated symbols, calculated lengths, DSECTs, etc.)  
+    1. Remove the use of "magic numbers" 
+14. Use meaningful comments 
+15. Use meaningful labels 
+16. Use meaningful DS instructions 
+17. Do not specify BLKSIZE for files
+    1. This allows the system to default to the optimal block size  
+    2. It also allows you to override the block size in the JCL   
+18. Make code reentrant 
+    1. Either get your own storage or use an internal stack  mechanism 
+    2. Split the code and data areas in your code 
```

Comment:
'Split the code and data areas...' should be a separate point.

Reply:
Done.